Assets/Scripts/Hex/HexVertices.cs

```diff
@@ -0,0 +1,239 @@
+    /// </summary>
+    public static readonly Orientation PointyTopOrientation = new Orientation(
+        new FloatHex[] {
+            new FloatHex(Mathf.Sqrt(3.0f), Mathf.Sqrt(3.0f) / 2.0f),
+            new FloatHex(0.0f, 3.0f / 2.0f)
+        },
+        new Vector3[] {
+            new Vector3(Mathf.Sqrt(3.0f) / 3.0f, 0.0f, -1.0f / 3.0f),
+            new Vector3(0.0f, 0.0f, 2.0f / 3.0f)
+        },
+        0.5f
+    );
+    
+    /// <summary>
+    /// Calculates a HexTile's vertices in world coordinates given a Hex
+    /// position and type.
+    /// The HexTile's center will always be at element 0.
+    /// </summary>
+    /// <param name="pos">The Hex to calculate.</param>
+    /// <param name="hexType">The type of Hex to calculate. This determines
+    /// at which angle the HexTiles will rotated.</param>
+    /// <returns>A Vector3[] of length NumVertices containing the vertices
+    /// required to draw the HexTile.</returns>
+    /// <exception cref="InvalidEnumArgumentException">Attempting to use an
+    /// unsupported Hex type.</exception>
+    public static Vector3[] GetVertices(Hex pos, HexTypes hexType)
+    {
+        Orientation orientation;
+        Vector3[] vertices;
+        switch (hexType)
+        {
+            case HexTypes.FlatTop:
+                orientation = FlatTopOrientation;
+                break;
+            case HexTypes.PointyTop:
+                orientation = PointyTopOrientation;
+                break;
+            default:
+                throw new InvalidEnumArgumentException("Unsupported Hex type " + hexType);
+        }
+        vertices = new Vector3[NumVertices];
+        vertices[0] = HexToWorld(orientation, pos);
+        for (uint i = 1; i < vertices.Length; i++)
+        {
+            // CalcCornerOffset is 0 indexed, so subtract one from the index.
+            vertices[i] = vertices[0] + CalcCornerOffset(orientation, i - 1);
+        }
+
+        return vertices;
+    }
+
+    /// <summary>
+    /// Calculate the corner offset given the index of a corner.
+    /// Corners are 0 indexed, start at Orientation.StartAngle, and go
+    /// counterclockwise.
+    /// </summary>
+    /// <param name="orientation">The type of Hex to calculate. This determines
+    /// at which angle teh HexTiles will be rotated.</param>
+    /// <param name="index">The index of the corner to calculate. For example,
+    /// on a flat topped Hex, index 0 represents the far right corner and 4
+    /// represents the bottom left corner.</param>
+    /// <returns>A Vector3 representing the corner offset from the center of the
+    /// Hex.</returns>
+    public static Vector3 CalcCornerOffset(Orientation orientation, uint index)
+    {
+        // Calculate on which angle of the unit circle this corner of the Hex lies.
+        // This implementation is from https://www.redblobgames.com/grids/hexagons/implementation.html#hex-geometry.

```

Comment:
If this is how you're going to credit redblobgames, it's going to clutter up the code. You didn't even do it everywhere you used their code. Remove this and add a single credit in the readme or at the top of this class.

Reply:
Added a credits section to the readme